scenarios/kubernetes_e2e.py

```diff
@@ -257,14 +257,27 @@ def start(self, args):
         check_env(env, self.command, *args)
 
 
-def cluster_name(cluster, build):
+def cluster_name(cluster):
     """Return or select a cluster name."""
     if cluster:
         return cluster
-    # Create a suffix based on the build number. Append a random string to it for
-    # avoiding potential conflicts across different jobs' runs (see issue #7592).
-    suffix = build if len(build) < 10 else hashlib.md5(build).hexdigest()[:10]
-    return 'e2e-%s-%s' % (suffix, os.urandom(3).encode('hex'))
+    # Create a suffix based on the build number and job name.
+    # This ensures no conflict across runs of different jobs (see #7592).
+    # For PR jobs, we use PR number instead of build number to ensure the
+    # name is constant across different runs of the presubmit on the PR.
+    # This helps clean potentially leaked resources from earlier run that
+    # could've got evicted midway (see #7673).
+    job_type = os.getenv('JOB_TYPE')
+    if job_type == 'batch':
+        suffix = 'batch-%s' % os.getenv('BUILD_ID', 0)
+    elif job_type == 'presubmit':
```

Comment:
why aren't we just using BUILD_ID for all cases?

Reply:
Er so we're going to depend on cluster naming for the same project + PR number + job and the side effect that kubetest will tear it down ... ? :hushed: 
This feels like a brittle work around that will come to bite us. In the future this will get refactored... We don't want to need the "scenarios" long term..